### PR TITLE
Buc postsubmit job

### PR DESCRIPTION
**Pull request by aszecowka**
- postsubmit job for buc
- add tests for validation yaml definitions (+ small common library)
- added presubmit job for test-infra to run tests mentioned previously

### CODE REVIEW

File: prow/jobs_test/ui-api-layer_test.go [hunk 1]
```diff
@@ -49,18 +39,7 @@ func TestUiApiLayerJobs(t *testing.T) {
...
-	}, master.Labels)
-
-	assert.Equal(t, map[string]string{
-		"preset-dind-enabled":           "true",
-		"preset-sa-gcr-push":            "true",
-		"preset-docker-push-repository": "true",
-		"preset-build-release":          "true",
-	}, release.Labels)
+	tester.AssertThatHasPresets(t, master.JobBase, tester.PresetDindEnabled, tester.PresetGcrPush, tester.PresetDockerPushRepo, tester.PresetBuildPr)
+	tester.AssertThatHasPresets(t, release.JobBase, tester.PresetDindEnabled, tester.PresetGcrPush, tester.PresetDockerPushRepo, tester.PresetBuildRelease)
```
Comment (aszecowka) [inline prow/jobs_test/ui-api-layer_test.go]: provide job that run tests!

File: development/tools/jobs/tester/tester.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+const (
+	PresetDindEnabled    Preset = "preset-dind-enabled"
+	PresetGcrPush        Preset = "preset-sa-gcr-push"
+	PresetDockerPushRepo Preset = "preset-docker-push-repository"
+	PresetBuildPr        Preset = "preset-build-pr"
+	PresetBuildMaster    Preset = "preset-build-master"
+	PresetBuildRelease   Preset = "preset-build-release"
+
+	ImageGolangBuildpack = "eu.gcr.io/kyma-project/prow/buildpack-golang:0.0.1"
```
Comment (michal-hudy) [inline development/tools/jobs/tester/tester.go]: Use a newer image.

File: prow/jobs/test-infra/test-jobs-yaml-definitions.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    branch: master
+    skip_report: false
+    decorate: true
+    path_alias: github.com/kyma-project/test-infra
+    spec:
+      containers:
+      - image: eu.gcr.io/kyma-project/prow/test-infra/buildpack-golang:v20181119-afd3fbd
+        workingDir: "/home/prow/go/src/github.com/kyma-project/test-infra/development/tools/jobs/"
+        env:
+        - name: GOPATH
```
Comment (michal-hudy) [inline prow/jobs/test-infra/test-jobs-yaml-definitions.yaml]: Is it not defined by default? If not, it should be defined in buildpack as decorator always checkout to `home/prow/go/src...`.

Comment (aszecowka) [reply prow/jobs/test-infra/test-jobs-yaml-definitions.yaml]: it should be, I will remove this env definition

**Pull request merged**